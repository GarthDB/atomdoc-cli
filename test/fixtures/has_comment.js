/**
 *  Public: parses a string representing an array into an Array.
 *
 *  * `val` {String} list string.
 *
 *  Returns {Array}
 */
function list(val) {
  const result = val.replace(/(^(?:"|')?\[)|(](?:"|')?$)/g, '').split(',');
  // This is just a regular comment.
  return result.map((item) => item.trim().replace(/(^(?:"|')?)|((?:"|')?$)/g, ''));
}
