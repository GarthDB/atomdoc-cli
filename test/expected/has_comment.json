[
  {
    "originalText": "Public: parses a string representing an array into an Array.\n\n* `val` {String} list string.\n\nReturns {Array}",
    "visibility": "Public",
    "description": "parses a string representing an array into an Array.",
    "summary": "parses a string representing an array into an Array.",
    "arguments": [
      {
        "name": "val",
        "description": "{String} list string.",
        "type": "String",
        "isOptional": false
      }
    ],
    "returnValues": [
      {
        "type": "Array",
        "description": "Returns {Array}"
      }
    ]
  },
  {
    "originalText": "This is just a regular comment.",
    "visibility": "Private",
    "description": "This is just a regular comment.",
    "summary": "This is just a regular comment."
  }
]
